tests/multipleBackend/objectPut.js

```diff
@@ -71,74 +74,99 @@ function put(bucketLoc, objLoc, requestHost, cb, errorDescription) {
+        {
+            name: 'Azure based on bucket location',
+            bucketLoc: 'azurebackend',
+            objLoc: null,
+        },
+        {
+            name: 'us-east-1 which is file based on bucket location if no locationConstraint provided',
+            bucketLoc: null,
+            objLoc: null,
+        },
+    ];
+
+    if (!isCEPH) {
+        putCases.push({
+            name: 'sproxyd',
+            bucketLoc: sproxydLocation,
+            objLoc: null,
         });
-    });
-
-    it('should put an object to file', done => {
-        put(memLocation, fileLocation, 'localhost', () => {
-            assert.deepStrictEqual(ds, []);
-            done();
-        });
-    });
-
-    it('should put an object to AWS', done => {
-        put(memLocation, 'awsbackend', 'localhost', () => {
-            assert.deepStrictEqual(ds, []);
-            done();
-        });
-    });
-
-    it('should put an object to mem based on bucket location', done => {
-        put(memLocation, null, 'localhost', () => {
-            assert.deepStrictEqual(ds[1].value, body);
-            done();
-        });
-    });
-
-    it('should put an object to file based on bucket location', done => {
-        put(fileLocation, null, 'localhost', () => {
-            assert.deepStrictEqual(ds, []);
-            done();
-        });
-    });
+    }
 
-    it('should put an object to AWS based on bucket location', done => {
-        put('awsbackend', null, 'localhost', () => {
-            assert.deepStrictEqual(ds, []);
-            done();
-        });
-    });
+    function isDataStoredInMem(testCase) {
+        return testCase.objLoc === memLocation
+               || (testCase.objLoc === null && testCase.bucketLoc === memLocation);
+    }
 
-    it('should put an object to Azure based on bucket location', done => {
-        put('azurebackend', null, 'localhost', () => {
+    function checkPut(testCase) {
+        if (isDataStoredInMem(testCase)) {
+            assert.deepStrictEqual(ds[ds.length - 1].value, body);
+        } else {
             assert.deepStrictEqual(ds, []);
-            done();
-        });
-    });
+        }
+    }
 
-    it('should put an object to Azure based on object location', done => {
-        put(memLocation, 'azurebackend', 'localhost', () => {
-            assert.deepStrictEqual(ds, []);
-            done();
-        });
+    afterEach(() => {
+        cleanup();
     });
 
-    it('should put an object to us-east-1 which is file based on bucket' +
-    ' location if no locationConstraint provided', done => {
-        put(null, null, 'localhost', () => {
-            assert.deepStrictEqual(ds, []);
-            done();
+    putCases.forEach(testCase => {
+        it(`should put an object to ${testCase.name}`, done => {
+            async.series([
+                next => put(testCase.bucketLoc, testCase.objLoc, 'localhost', () => {
+                    checkPut(testCase);
+                    next();
+                }),
```

Comment:
nit:
```suggestion
                next => put(testCase.bucketLoc, testCase.objLoc, 'localhost', next),
                next => {
                    checkPut(testCase);
                    next();
                }),
```

(and we may then merge it with the next `setTimeout` to simplify further)